Mathlib/Topology/UrysohnsLemma.lean

```diff
@@ -465,3 +468,245 @@ theorem exists_continuous_nonneg_pos [RegularSpace X] [LocallyCompactSpace X] (x
   have := fk (mem_of_mem_nhds k_mem)
   simp only [ContinuousMap.coe_mk, Pi.one_apply] at this
   simp [this]
+
+/-- A variation of Urysohn's lemma. In a normal space `X`, for a closed set `t` and an open set `s`
+such that `t ⊆ s`, there is a continuous function `f` supported in `s`, `f x = 1` on `t` and
+`0 ≤ f x ≤ 1`. -/
+lemma exists_tsupport_one_of_isOpen_isClosed [NormalSpace X] {s t : Set X}
+    (hs : IsOpen s) (ht : IsClosed t) (hst : t ⊆ s) : ∃ f : C(X, ℝ), tsupport f ⊆ s ∧ EqOn f 1 t
+    ∧ ∀ x, f x ∈ Icc (0 : ℝ) 1 := by
+    obtain ⟨U, V, hUV⟩ := normal_separation (IsOpen.isClosed_compl hs) ht
+      (HasSubset.Subset.disjoint_compl_left hst)
+    have hDisjoint : Disjoint (closure U) t := by
+      apply le_compl_iff_disjoint_right.mp
+      apply _root_.subset_trans _ (Set.compl_subset_compl.mpr hUV.2.2.2.1)
+      apply (IsClosed.closure_subset_iff (IsOpen.isClosed_compl hUV.2.1)).mpr
+      exact Set.subset_compl_iff_disjoint_right.mpr hUV.2.2.2.2
+    obtain ⟨f, hf⟩ := exists_continuous_zero_one_of_isClosed isClosed_closure ht hDisjoint
+    use f
+    constructor
+    · apply _root_.subset_trans _ (Set.compl_subset_comm.mp hUV.2.2.1)
+      rw [← IsClosed.closure_eq (IsOpen.isClosed_compl hUV.1)]
+      apply closure_mono
+      rw [Set.subset_compl_iff_disjoint_left, Function.disjoint_support_iff]
+      exact Set.EqOn.mono subset_closure hf.1
+    · exact hf.2
+
+open Classical BigOperators
+
+/-- A variation of Urysohn's lemma. In a normal T2 space `X`, for a compact set `t` and a finite
+family of open sets `{s i}` such that `t ⊆ ⋃ i, s i`, there is a family of continuous function
+`{f i}` supported in `s i`, `∑ i, f i x = 1` on `t` and `0 ≤ f x ≤ 1`. -/
+lemma exists_forall_tsupport_iUnion_one_iUnion_of_isOpen_isClosed [NormalSpace X] [T2Space X]
```

Comment:
More importantly: how does this relate to the results about partitions of unity, e.g. `exists_isSubordinate_of_locallyFinite` in `Topology.PartitionOfUnity`? That seems more general.

Reply:
thank you for your review!
sorry that it took me a while and that the intention of this PR wasn't unclear. Here I wanted to remove `T4Space` (`NormalSpace`) using #12459. `exists_isSubordinate_of_locallyFinite` in `Topology.PartitionOfUnity` depends on `NormapSpace`, so I cannot do it usisng partition of unity. Does this make sense? If so, I continue editing this PR.

Reply:
Ah, if you weaken the assumptions here to `T2Space` then that is valuable.

Reply:
While the proof of `BumpCovering.exists_isSubordinate_of_locallyFinite_of_prop` needs `NormalSpace X` to get a subcovering with some properties, you can use the rest of the results (e.g., `BumpCovering.toPartitionOfUnity`) to simplify your proof. BTW, is `[R1Space X]` enough or you need `[T2Space X]`?

Reply:
Also, please reformulate the result as existence of a `PartitionOfUnity` subordinate to your covering. This will work much better with the rest of the library.

Reply:
@urkud I tried to use `PartitionOfUnity` (in the file `PartitionOfUnity.lean`), but even the trivial case gets very long, I am sure this is not what you intended but I don't know how to do it. Could you help? Do you mean to prove basic lemmas using `R1Space` `LocallyCompactSpace` instead of `NormalSpace` `ParacompactSpace`?

Reply:
I'm going to have a look tomorrow.